scripts/cc-cli/src/test/integration-tests/collectCoins.test.ts

```diff
@@ -0,0 +1,138 @@
+    setArg(
+      "CREDITCOIN_CTC_CONTRACT_ADDRESS",
+      process.env.CREDITCOIN_CTC_CONTRACT_ADDRESS,
+    );
+    setArgIfUndef(
+      "CREDITCOIN_CTC_BURN_TX_HASH",
+      process.env.CREDITCOIN_CTC_BURN_TX_HASH,
+    );
+
+    await retry(5, async (resolve: any, reject: any) => {
+      await setAuthorities().then(resolve).catch(reject);
+    }).catch(() => {
+      console.log("Could not setup testing authorities");
+      process.exit(1);
+    });
+
+    await cryptoWaitReady();
+
+    caller = randomTestAccount(false);
+
+    ccApi = await creditcoinApi((global as any).CREDITCOIN_API_URL);
+    sudo = arg("CREDITCOIN_CREATE_SIGNER")(keyring, "lender");
+
+    const { api } = ccApi;
+
+    /* eslint-disable @typescript-eslint/naming-convention */
+    const contract = api.createType(
+      "PalletCreditcoinOcwTasksCollectCoinsDeployedContract",
+      {
+        address: (global as any).CREDITCOIN_CTC_CONTRACT_ADDRESS,
+        chain: blockchain,
+      },
+    );
+
+    await api.tx.sudo
+      .sudo(api.tx.creditcoin.setCollectCoinsContract(contract))
+      .signAndSend(sudo, { nonce: -1 });
+  }, 100_000);
+
+  afterAll(async () => {
+    await ccApi.api.disconnect();
+  });
+
+  test("e2e", async () => {
+    const { api } = ccApi;
+
+    const provider = new providers.JsonRpcProvider(
+      arg("CREDITCOIN_ETHEREUM_NODE_URL"),
+    );
+    const deployerWallet = new Wallet(
+      arg("CREDITCOIN_CTC_DEPLOYER_PRIVATE_KEY"),
+      provider,
+    );
+
+    const fundTx = await fundFromSudo(
+      caller.address,
+      parseAmountInternal("1000000"),
+      arg("CREDITCOIN_API_URL"),
+    );
+    await signSendAndWatch(fundTx, api, sudo);
+
+    const registerResult = execa.commandSync(
+      `npx creditcoin-cli register-address -u ${
+        arg("CREDITCOIN_API_URL") as string
+      }`,
+      {
+        env: {
+          BLOCKCHAIN: "Ethereum",
+          PRIVATE_KEY: arg("CREDITCOIN_CTC_DEPLOYER_PRIVATE_KEY"),
+          CC_SECRET: caller.secret,
+        },
+      },
+    );
+
+    const stdout = registerResult.stdout.split("\n");
+    expect(registerResult.failed).toBe(false);
+    expect(registerResult.exitCode).toBe(0);
+    expect(registerResult.stderr).toBe("");
+    expect(
+      stdout[stdout.length - 1].includes("Address Registered Successfully"),
+    ).toBe(true);
+
+    const collectResult = execa.commandSync(
+      `npx creditcoin-cli collect-coins -u ${
+        arg("CREDITCOIN_API_URL") as string
+      }`,
+      {
+        env: {
+          EXTERNAL_ADDR: deployerWallet.address,
+          BURN_TX_HASH: arg("CREDITCOIN_CTC_BURN_TX_HASH"),
+          CC_SECRET: caller.secret,
+        },
+      },
+    );
+
+    const collectOutput = collectResult.stdout.split("\n");
+    expect(collectResult.failed).toBe(false);
+    expect(collectResult.exitCode).toBe(0);
+    expect(collectResult.stderr).toBe("");
+    expect(collectOutput[collectOutput.length - 1]).toBe("Success!");
```

Comment:
Can you assert on the available balance before and after collect coins ?

Reply:
`api.query.system.account` also shows values that don't make sense...

Reply:
> `api.query.system.account` also shows values that don't make sense...

Anything in particular? 

I'm not able to find documentation for this query, only found this https://polkadot.js.org/docs/api/FAQ#my-chain-does-not-support-systemaccount-queries, so I'm assuming the balances don't look right.


Maybe you can use `getBalance()`, see https://polkadot.js.org/docs/substrate/rpc/#getbalanceaddress-h160-number-blocknumber-u256. 

If you can push some WIP code of that you are trying to do I will take a look.

Reply:
I just realized we can also use `getBalance()` from cc-cli, see https://github.com/gluwa/creditcoin/blob/dev/scripts/cc-cli/src/utils/balance.ts

Reply:
Using the `api.query.system.account` method showed a difference of 0. 

The `getBalance` command also showed zero. I pushed the changes that show how I am making those calls and asserting on them

Reply:
In addition to the method tried in the latest push I also tried something similar to this `expect(ending.total.sub(starting.total).eq(new BN(1))).toBe(true);` without success